sdk/core/azure-core/src/main/java/com/azure/core/util/tracing/TracerSpanAttributes.java

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.tracing;
+
+/**
+ * Class to hold the attributes for tracer spans.
+ */
+public class TracerSpanAttributes {
+    private final String resourceProviderName;
```

Comment:
Since this has just one attribute now, I am assuming new attributes will be added later. If that's the intent, then shouldn't we make this a fluent API? A public constructor with just one attribute now will result in breaking changes if we need more attributes here later. 

Alternatively, do we need this attributes class, if all fields in here are going to be `String` types? Can we instead use `Map<String, String>` as properties bag?

Reply:
Span attributes coming from clients would be expected to be strings and hence updating it to use a `Map<String, String>` instead.